maze-game/MyWorld.java

```diff
@@ -153,62 +160,118 @@ private void addGhost4()
-        int spawn_point = (Greenfoot.getRandomNumber(11));
-        
-        if (spawn_point == 0)
-        {
-            addObject(new WallBreaker(), 25, 375);
-        }
+        int[] result = new int[2];
         if (spawn_point == 1)
         {
-            addObject(new WallBreaker(), 125, 675);
+            result[0] = 25;
+            result[1] = 375;
+            return(result);
         }
         if (spawn_point == 2)
         {
-            addObject(new WallBreaker(), 125, 775);
+            result[0] = 125;
+            result[1] = 675;
+            return(result);
         }
         if (spawn_point == 3)
         {
-            addObject(new WallBreaker(), 225, 775);
+            result[0] = 125;
+            result[1] = 775;
+            return(result);
         }
         if (spawn_point == 4)
         {
-            addObject(new WallBreaker(), 575, 325);
+            result[0] = 225;
+            result[1] = 775;
+            return(result);
         }
         if (spawn_point == 5)
         {
-            addObject(new WallBreaker(), 575, 775);
+            result[0] = 575;
+            result[1] = 325;
+            return(result);
         }
         if (spawn_point == 6)
         {
-            addObject(new WallBreaker(), 675, 775);
+            result[0] = 575;
+            result[1] = 775;
+            return(result);
         }
         if (spawn_point == 7)
         {
-            addObject(new WallBreaker(), 775, 175);
+            result[0] = 675;
+            result[1] = 775;
+            return(result);
         }
         if (spawn_point == 8)
         {
-            addObject(new WallBreaker(), 775, 475);
+            result[0] = 775;
+            result[1] = 175;
+            return(result);
         }
         if (spawn_point == 9)
         {
-            addObject(new WallBreaker(), 875, 175);
+            result[0] = 775;
+            result[1] = 475;
+            return(result);
         }
         if (spawn_point == 10)
         {
-            addObject(new WallBreaker(), 1125, 75);
+            result[0] = 875;
+            result[1] = 175;
+            return(result);
         }
         if (spawn_point == 11)
         {
-            addObject(new WallBreaker(), 1125, 475);
+            result[0] = 1125;
+            result[1] = 75;
+            return(result);
+        }
+        if (spawn_point == 12)
+        {
+            result[0] = 1125;
+            result[1] = 475;
+            return(result);
+        }
+        return(result);
+    }
+    //random number generator that excludes numbers
+    public int getRandomWithExclusion(Random rnd, int start, int end, int[] exclude) 
+    {
+        int random = start + rnd.nextInt(end - start + 1 - exclude.length);
+        for (int ex : exclude) {
+            if (random < ex) {
+                break;
+            }
+            random++;
         }
+        return random;//somehow they stay on seperate sides FIX THIS
```

Comment:
has this been fixed? does it need fixing?

Reply:
its relatively fixed I don't think it needs too much fixing its not that noticeable if it is broken